styx-cli/src/main/java/com/spotify/styx/cli/BackfillCli.java

```diff
@@ -0,0 +1,226 @@
+  private static final String START = "start";
+  private static final String END = "end";
+
+  static Subparsers registerCommands(Subparsers subparsers) {
+    final Subparsers backfillParser = Main.Command.BACKFILL.parser(subparsers)
+            .addSubparsers().title("commands").metavar(" ");
+
+    final Subparser show = BackfillCommand.SHOW.parser(backfillParser);
+    show.addArgument("backfill").dest(ID).help("Backfill ID");
+
+    final Subparser edit = BackfillCommand.EDIT.parser(backfillParser);
+    edit.addArgument("backfill").help("Backfill ID");
+    edit.addArgument("--concurrency").dest(CONCURRENCY)
+        .help("set the concurrency value for the backfill")
+        .type(Integer.class);
+
+    final Subparser halt = BackfillCommand.HALT.parser(backfillParser);
+    halt.addArgument("backfill").help("Backfill ID");
+
+    final Subparser list = BackfillCommand.LIST.parser(backfillParser);
+    list.addArgument("-w", "--workflow").dest(WORKFLOW)
+        .help("only show backfills for WORKFLOW");
+    list.addArgument("-c", "--component").dest(COMPONENT)
+        .help("only show backfills for COMPONENT");
+    list.addArgument("-a", "--show-all").dest(SHOW_ALL)
+        .setDefault(false)
+        .action(Arguments.storeTrue())
+        .help("show all backfills, even halted and all-triggered ones");
+
+    final Main.PartitionAction partitionAction = new Main.PartitionAction();
+    final Subparser create = BackfillCommand.CREATE.parser(backfillParser);
+    create.addArgument(COMPONENT)
+        .help("Component ID");
+    create.addArgument(WORKFLOW)
+        .help("Workflow ID");
+    create.addArgument(START)
+        .help("Start date/datehour (inclusive)")
+        .action(partitionAction);
+    create.addArgument(END)
+        .help("End date/datehour (exclusive)")
+        .action(partitionAction);
+    create.addArgument(CONCURRENCY)
+        .help("The number of jobs to run in parallel")
+        .type(Integer.class);
+
+    return backfillParser;
+  }
+
+  enum BackfillCommand {
+    LIST("ls", "List active backfills. Use option -a (--show-all) to show all"),
+    CREATE("", "Create a backfill"),
+    EDIT("e", "Edit a backfill"),
+    HALT("h", "Halt a backfill"),
+    SHOW("get", "Show info about a specific backfill");
+
+    private final String alias;
+    private final String description;
+
+    BackfillCommand(String alias, String description) {
+      this.alias = alias;
+      this.description = description;
+    }
+
+    public Subparser parser(Subparsers subCommands) {
+      final Subparser subparser = subCommands
+          .addParser(name().toLowerCase())
+          .setDefault(Main.SUBCOMMAND_DEST, this)
+          .description(description)
+          .help(description);
+
+      if (alias != null && !alias.isEmpty()) {
+        subparser.aliases(alias);
+      }
+
+      return subparser;
+    }
+  }
+
+  static void runCommand(Main main, BackfillCommand backfillCommand)
```

Comment:
Can we eliminate the circular dependency between `Main` and `{Backfill, Resource}Cli`?